sequencer/src/api/data_source.rs

```diff
@@ -134,6 +142,71 @@ pub(crate) trait CatchupDataSource {
 
 impl CatchupDataSource for MetricsDataSource {}
 
+pub struct MyHotShotConfig(HotShotConfig<PubKey>);
+
+impl MyHotShotConfig {
+    pub fn new(c: HotShotConfig<PubKey>) -> Self {
+        Self(c)
+    }
+}
+pub struct MyValidatorConfig<'a>(&'a ValidatorConfig<PubKey>);
+
+impl<'a> Serialize for MyValidatorConfig<'a> {
+    fn serialize<S>(&self, serializer: S) -> Result<S::Ok, S::Error>
+    where
+        S: serde::Serializer,
+    {
+        let mut state = serializer.serialize_struct("MyValidatorConfig", 3)?;
+
+        state.serialize_field("public_key", &self.0.public_key)?;
+        state.serialize_field("stake_value", &self.0.stake_value)?;
+        state.serialize_field("is_da", &self.0.is_da)?;
+
+        state.end()
+    }
+}
+
+impl Serialize for MyHotShotConfig {
```

Comment:
can't think of a better way to skip field without adding it in hotshot

Reply:
I think ideally sensitive data would have a Display implementation that prints a redacted version of it.

Reply:
Or I guess that wouldn't even help here. Does it even make sense to use `Serialize` here? I think serialization usually implies that deserialization is the inverse which won't work anyway if we hide some fields.

Reply:
shouldn't we skip serialization for these fields in hotshot types?

Reply:
but then we load the config from the storage when starting up the node

Reply:
Yeah, I think the solution to *that* problem is storing the config encrypted in the db/file system. But IMO that's much less critical than exposing private keys in plaintext *in API endpoints*, so having a solution specifically for the latter seems good for now.

FWIW I think this approach is fine. And the serialization here is bidirectional, because the "hiding" of the private keys happens on creation of the `MyHotShotConfig`: we can successfully round trip a `MyHotShotConfig`, hidden fields and all, we just can't get back a `HotShotConfig` from it, which is fine.

Reply:
My only concerns:
* by specifically selecting which fields to hide, it seems very easy to accidentally miss something and leak secret data over a public API. While it is going to be messier, I think it would be safer to take the opposite approach and specifically select the fields we want to include. If we do that, we might as well just make it a proper struct with all the fields we want from the `HotShotConfig`, instead of a `json::Value` wrapper
* We should call it `PublicHotShotConfig` or something, `My` doesn't mean anything